packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx

```diff
@@ -29,31 +54,49 @@ import CardHeader from '../../../components/CardHeader';
 export const viewName = 'account-recovery-reset-password';
 
 export type AccountRecoveryResetPasswordProps = {
-  email: string;
-  linkStatus: LinkStatus;
-  forceAuth?: boolean;
-};
+  overrides?: {
+    navigate?: NavigateFn;
+    locationContext?: GenericContext;
+    urlSearchContext?: UrlSearchContext;
+  };
+} & RouteComponentProps;
 
 type FormData = {
   newPassword: string;
   confirmPassword: string;
 };
 
-type LinkStatus = 'expired' | 'damaged' | 'valid';
-
 const AccountRecoveryResetPassword = ({
-  email,
-  linkStatus,
-  forceAuth = false,
-}: AccountRecoveryResetPasswordProps & RouteComponentProps) => {
+  overrides,
+}: AccountRecoveryResetPasswordProps) => {
   usePageViewEvent(viewName, REACT_ENTRYPOINT);
 
-  const [passwordMatchErrorText, setPasswordMatchErrorText] =
-    useState<string>('');
+  // Grab state from hooks
   const alertBar = useAlertBar();
-  const navigate = useNavigate();
+  const notifier = useNotifier();
+  const broker = useBroker();
   const ftlMsgResolver = useFtlMsgResolver();
+  const account = useAccount();
+  const relier = useRelier();
+  let navigate = useNavigate();
+  let urlSearchContext = useUrlSearchContext();
+  let locationContext = useLocationContext();
+
+  // Apply overrides
+  navigate = overrides?.navigate || navigate;
+  urlSearchContext = overrides?.urlSearchContext || urlSearchContext;
+  locationContext = overrides?.locationContext || locationContext;
 
+  const verificationInfo = new VerificationInfo(urlSearchContext);
```

Comment:
@millsoper Since the VerificationInfo model is so simple no factory is needed. We can just pass in the urlSearchContext, which is available on the AppContext.